src/components/UserDetail.jsx

```diff
@@ -34,6 +39,34 @@ export default class UserDetail extends React.Component {
       console.log('loading comments for user ' + nextProps.selectedUser._id);
       nextProps.dispatch(fetchCommentsByUser(nextProps.selectedUser._id));
     }
+    this.updateTagsList(nextProps.selectedUser);
+  }
+
+  updateTagsList(user) {
+    this.tags = [];
```

Comment:
I'd probably change the function to something like `getTags()` and call it directly from line 82
`tags={this.getTags(this.props.selectedUser)}`